katas/content/superposition/wstate_power_of_two/solution.md

```diff
@@ -0,0 +1,48 @@
+The problem becomes more manageable if broken down into the simplest cases and built up from there.
+
+1. The smallest instance of the problem, $N = 1$, requires preparing $|W_1\rangle = |1\rangle$; this can be done trivially using an X gate.
+
+2. The next instance, $N = 2$, requires preparing $|W_2\rangle = \frac{1}{\sqrt2}\big(|10\rangle + |01\rangle\big)$. It matches one of the Bell states we've seen earlier, but preparing it will be more interesting (and more useful for the next steps!) if we think of it in recursive terms. Let's see how to express $|W_2\rangle$ in terms of $|W_1\rangle$:
+
+$$|W_2\rangle = \frac{1}{\sqrt2}\big(|10\rangle + |01\rangle\big) = \frac{1}{\sqrt2}\big(|W_1\rangle \otimes |0\rangle + |0\rangle \otimes |W_1\rangle\big)$$
+
+This representation suggests us a solution: "split" the starting state $|00\rangle$ in two terms, prepare $|W_1\rangle$ on the first qubit for the first term and on the second qubit - for the second term.
+To do this, we can again use an auxiliary qubit prepared in the $|+\rangle$ state and control the preparation of $|W_1\rangle$ state on the first or the second qubit based on the state of the auxiliary qubit:
+
+$$|0\rangle_{aux} |00\rangle_{reg} \overset{H}{\longrightarrow}\frac{1}{\sqrt2}(|0\rangle + |1\rangle)_{aux} \otimes |00\rangle_{reg} = \frac{1}{\sqrt2}(|0\rangle_{aux} |00\rangle_{reg} + |1\rangle_{aux} |00\rangle_{reg})\overset{CNOT_0}{\longrightarrow}$$
```

Comment:
I have been struggling with getting this formatted correctly, something is still wrong though I'm not sure where

Reply:
This change fixed it, but needs checking it's still equivalent: https://github.com/microsoft/qsharp/pull/1457/commits/821913a601913a2a929af0bd425b38165eac395f

Reply:
Looks great to me, though I'm looking at the finished version already :-)